src/components/pages/Desk/__specs__/Desk.spec.jsx

```diff
@@ -270,6 +270,29 @@ describe('src | components | Desk', () => {
       const responseFromApi = await screen.findByText('error message')
       expect(responseFromApi).toBeInTheDocument()
     })
+
+    it('should display an error message when the booking invalidation has failed', async () => {
+      // given
+      jest.spyOn(props, 'getBooking').mockRejectedValue({
+        json: jest.fn(() => Promise.resolve({ booking: 'token is already validated' })),
+        status: 410,
+      })
+      jest.spyOn(props, 'invalidateBooking').mockRejectedValue({
+        json: jest.fn(() => Promise.resolve({ booking: 'cannot invalidate booking' })),
+        status: 403,
+      })
+      renderDesk(props)
+      const tokenInput = screen.getByLabelText('Contremarque')
+      fireEvent.change(tokenInput, { target: { value: 'MEFA01' } })
+      const submitButton = await screen.findByRole('button', { name: 'Invalider la contremarque' })
+
+      // when
+      fireEvent.click(submitButton)
+
+      // then
+      const responseFromApi = await screen.findByText('cannot invalidate booking')
```

Comment:
getByText te permet de vérifier qu'il y est :)
```suggestion
      await screen.getByText('cannot invalidate booking')
```

Reply:
https://testing-library.com/docs/queries/about/#types-of-queries

Idem pour findByRole

Reply:
J'ai (bêtement) recopié le test précédent. J'hésite à changer de façon de faire juste pour ce test, du coup. Et j'hésite encore plus à ré-écrire les autres tests avec les fonctions que tu proposes. Quel est ton avis ?

Reply:
La pratique sur pro est d'utiliser `get` si tu es certain d'avoir l'élément, mais de tout de même faire un assert dessus (`toBeInDocument`)
Le get jettera une exception plus parlante en cas d'absence
Et le assert montre clairement ce qu'on teste (sinon on galère à trouver l'assertion qui est en fait qu'on attend que get ne nous jette pas)

donc modifie le à minima sur ce test

Par contre `get` ne s'await pas, cf https://testing-library.com/docs/react-testing-library/cheatsheet

Reply:
Vu sur Slack : ici, le texte à chercher est affiché via une promesse, on doit donc attendre, ce que ne fait pas `getByText()`. Je vais donc rester sur `await screen.findByText()`.